getcontext/tracing/trace.py

```diff
@@ -128,19 +128,28 @@ def _parse_evaluation_results(self, results: EvaluationsRunResponse):
         for result in results.results:
             test_case_name = result.test_case.name
             for evaluation in result.evaluations:
-                tick = u'\u2705'
-                x_mark = u'\u274C'
-                symbol = tick if evaluation.outcome == "positive" else x_mark
                 msg = (
-                    f"{symbol} {test_case_name}: Evaluation {outcome_map.get(evaluation.outcome)} for"
-                    f" {evaluation.evaluator_name}:\n\t {evaluation.reasoning}"
+                    f"{self._symbol(evaluation.outcome == 'positive')} {test_case_name}: "
+                    f"Evaluation {outcome_map.get(evaluation.outcome)} for "
+                    f"{evaluation.evaluator_name}: {self._create_reasoning_msg(evaluation.reasoning)}"
                 )
-                if evaluation.outcome == "positive":
-                    print(f"\n{msg}")
-                else:
+                # TODO: use correct logger here
+                print(f"\n{msg}")
+
+                if evaluation.outcome != "positive":
                     failed_evaluation_msgs.append(msg)
         return failed_evaluation_msgs
 
+    def _create_reasoning_msg(self, reasoning: EvaluationReasoning):
+        if reasoning is None:
+            return ""
+
+        return "".join([f"\n\t{self._symbol(t.verdict)} {t.reason}" for t in reasoning.result])
```

Comment:
This is really tricky to read, but I don't know enough idiomatic Python to suggest an alternative. Can you see anything that can be done here?

Reply:
Ironically I might have gone over the top on Python functionality here, let me take another pass.